diff -NEur '--exclude-from=/home/tango/nmp_4x86/scripts/diff-exclude' neutrino-mp.org/src/gui/movieplayer.h neutrino-mp/src/gui/movieplayer.h
--- neutrino-mp.org/src/gui/movieplayer.h	2018-10-05 23:17:50.842340908 +0200
+++ neutrino-mp/src/gui/movieplayer.h	2018-10-05 23:29:33.933229632 +0200
@@ -32,6 +32,8 @@
 #ifndef __movieplayergui__
 #define __movieplayergui__
 
+#define MAX_PLAYBACK_PIDS 40
+
 #include <config.h>
 #include <configfile.h>
 #include <gui/filebrowser.h>
diff -NEur '--exclude-from=/home/tango/nmp_4x86/scripts/diff-exclude' neutrino-mp.org/src/gui/vfd_setup.cpp neutrino-mp/src/gui/vfd_setup.cpp
--- neutrino-mp.org/src/gui/vfd_setup.cpp	2018-10-05 23:17:50.846340831 +0200
+++ neutrino-mp/src/gui/vfd_setup.cpp	2018-10-05 23:32:27.658023209 +0200
@@ -127,7 +127,7 @@
 		vfds->addItem(mf);
 	}
 
-	if (g_info.hw_caps->can_set_display_brightness)
+	if (g_info.hw_caps->display_can_set_brightness)
 	{
 		//vfd brightness menu
 		mf = new CMenuForwarder(LOCALE_LCDMENU_LCDCONTROLER, vfd_enabled, NULL, this, "brightness", CRCInput::RC_green);
diff -NEur '--exclude-from=/home/tango/nmp_4x86/scripts/diff-exclude' neutrino-mp.org/src/zapit/include/zapit/capmt.h neutrino-mp/src/zapit/include/zapit/capmt.h
--- neutrino-mp.org/src/zapit/include/zapit/capmt.h	2017-03-14 13:49:59.576824828 +0100
+++ neutrino-mp/src/zapit/include/zapit/capmt.h	2018-10-05 23:25:45.081472051 +0200
@@ -66,7 +66,11 @@
 		bool sendMessage(const char * const data, const size_t length, bool update = false);
 		bool makeCaPmt(CZapitChannel * channel, bool add_private, uint8_t list = CAPMT_ONLY, const CaIdVector &caids = CaIdVector());
 		bool setCaPmt(bool update = false);
+#if ! HAVE_COOL_HARDWARE
+		bool sendCaPmt(uint64_t tpid, uint8_t *rawpmt, int rawlen, uint8_t type, unsigned char scrambled = 0, casys_map_t camap = std::set<int>(), int mode = 0 , bool enable = false);
+#else
 		bool sendCaPmt(uint64_t tpid, uint8_t *rawpmt, int rawlen, uint8_t type);
+#endif
 		int  makeMask(int demux, bool add);
 		int  getCaMask(void) { return camask; }
 		void setCaMask(int mask) { camask = mask; }
diff -NEur '--exclude-from=/home/tango/nmp_4x86/scripts/diff-exclude' neutrino-mp.org/src/zapit/src/capmt.cpp neutrino-mp/src/zapit/src/capmt.cpp
--- neutrino-mp.org/src/zapit/src/capmt.cpp	2017-03-14 13:49:59.588824734 +0100
+++ neutrino-mp/src/zapit/src/capmt.cpp	2018-10-05 23:24:31.462841620 +0200
@@ -142,11 +142,19 @@
 	return sendMessage((char *)cabuf, calen, update);
 }
 
+#if ! HAVE_COOL_HARDWARE
+bool CCam::sendCaPmt(uint64_t tpid, uint8_t *rawpmt, int rawlen, uint8_t type, unsigned char scrambled, casys_map_t camap, int mode, bool enable)
+{
+	return cCA::GetInstance()->SendCAPMT(tpid, source_demux, camask,
+			rawpmt ? cabuf : NULL, rawpmt ? calen : 0, rawpmt, rawpmt ? rawlen : 0, (CA_SLOT_TYPE) type, scrambled, camap, mode, enable);
+}
+#else
 bool CCam::sendCaPmt(uint64_t tpid, uint8_t *rawpmt, int rawlen, uint8_t type)
 {
 	return cCA::GetInstance()->SendCAPMT(tpid, source_demux, camask,
 			rawpmt ? cabuf : NULL, rawpmt ? calen : 0, rawpmt, rawpmt ? rawlen : 0, (CA_SLOT_TYPE) type);
 }
+#endif
 
 int CCam::makeMask(int demux, bool add)
 {
