diff -Ebur '--exclude=.svn' '--exclude=.git' neutrino-mp.org/src/eitd/sectionsd.cpp neutrino-mp/src/eitd/sectionsd.cpp
--- neutrino-mp.org/src/eitd/sectionsd.cpp	2014-02-14 22:46:49.000000000 +0100
+++ neutrino-mp/src/eitd/sectionsd.cpp	2014-03-03 16:47:23.801915836 +0100
@@ -1451,7 +1451,7 @@
 		dvb_time = 0;
 		timediff = 0;
 
-		if (ntpenable && system( ntp_system_cmd.c_str() ) == 0) {
+		if (ntpenable /*&& system( ntp_system_cmd.c_str() ) == 0*/) {
 			time_ntp = true;
 			success = true;
 		} else if (dvb_time_update) {
diff -Ebur '--exclude=.svn' '--exclude=.git' neutrino-mp.org/src/neutrino.cpp neutrino-mp/src/neutrino.cpp
--- neutrino-mp.org/src/neutrino.cpp	2014-03-03 16:39:31.909934424 +0100
+++ neutrino-mp/src/neutrino.cpp	2014-03-03 16:47:23.825915835 +0100
@@ -673,7 +673,7 @@
 	std::string timeshiftDir;
 	if(g_settings.timeshiftdir.empty()) {
 		timeshiftDir = g_settings.network_nfs_recordingdir + "/.timeshift";
-		safe_mkdir(timeshiftDir.c_str());
+		//safe_mkdir(timeshiftDir.c_str());
 	} else {
 		if(g_settings.timeshiftdir != g_settings.network_nfs_recordingdir)
 			timeshiftDir = g_settings.timeshiftdir;
@@ -2311,9 +2311,9 @@
 		saveSetup(NEUTRINO_SETTINGS_FILE);
 	}
 
-	CHDDDestExec * hdd = new CHDDDestExec();
-	hdd->exec(NULL, "");
-	delete hdd;
+	//CHDDDestExec * hdd = new CHDDDestExec();
+	//hdd->exec(NULL, "");
+	//delete hdd;
 fprintf(stderr, "[neutrino start] %d  -> %5ld ms\n", __LINE__, time_monotonic_ms() - starttime);
 
 	hintBox->hide(); // InitZapper also displays a hintbox
