diff -NEbur '--exclude-from=/home/tango/nmp_4x86/diff-exclude' neutrino-mp.org/src/eitd/sectionsd.cpp neutrino-mp/src/eitd/sectionsd.cpp
--- neutrino-mp.org/src/eitd/sectionsd.cpp	2014-11-24 23:19:40.092439603 +0100
+++ neutrino-mp/src/eitd/sectionsd.cpp	2014-11-24 23:20:03.188439394 +0100
@@ -1443,7 +1443,7 @@
 		dvb_time = 0;
 		timediff = 0;
 
-		if (ntpenable && system( ntp_system_cmd.c_str() ) == 0) {
+		if (ntpenable /*&& system( ntp_system_cmd.c_str() ) == 0*/) {
 			time_ntp = true;
 			success = true;
 		} else if (dvb_time_update) {
diff -NEbur '--exclude-from=/home/tango/nmp_4x86/diff-exclude' neutrino-mp.org/src/neutrino.cpp neutrino-mp/src/neutrino.cpp
--- neutrino-mp.org/src/neutrino.cpp	2014-11-24 23:19:40.108439603 +0100
+++ neutrino-mp/src/neutrino.cpp	2014-11-24 23:21:00.332438876 +0100
@@ -665,7 +665,7 @@
 	std::string timeshiftDir;
 	if(g_settings.timeshiftdir.empty()) {
 		timeshiftDir = g_settings.network_nfs_recordingdir + "/.timeshift";
-		safe_mkdir(timeshiftDir.c_str());
+		//safe_mkdir(timeshiftDir.c_str());
 	} else {
 		if(g_settings.timeshiftdir != g_settings.network_nfs_recordingdir)
 			timeshiftDir = g_settings.timeshiftdir;
@@ -2310,9 +2310,9 @@
 		saveSetup(NEUTRINO_SETTINGS_FILE);
 	}
 
-	CHDDDestExec * hdd = new CHDDDestExec();
-	hdd->exec(NULL, "");
-	delete hdd;
+	//CHDDDestExec * hdd = new CHDDDestExec();
+	//hdd->exec(NULL, "");
+	//delete hdd;
 
 	bootstatus->hide();
 	delete bootstatus;
diff -NEbur '--exclude-from=/home/tango/nmp_4x86/diff-exclude' neutrino-mp.org/src/gui/movieplayer.cpp neutrino-mp/src/gui/movieplayer.cpp
--- neutrino-mp.org/src/gui/movieplayer.cpp	2018-01-05 13:31:56.805244098 +0100
+++ neutrino-mp/src/gui/movieplayer.cpp	2018-02-11 14:16:03.116062693 +0100
@@ -152,10 +152,12 @@
 		delete bookmarkmanager;
 		bookmarkmanager = NULL;
 	}
+/*
 	if(playback){
 		delete playback;
 		playback = NULL;
 	}
+*/
 	if (this == instance_mp) {
 		delete instance_bg;
 		instance_bg = NULL;
